File: Data Structures & Algorithms/maximum-product-subarray/submission-10.cs
public class Solution {
    public int MaxProduct(int[] nums) {
        int res = nums[0];

        for (int i = 0; i < nums.Length; i++) {
            int cur = nums[i];
            res = Math.Max(res, cur);
            for (int j = i + 1; j < nums.Length; j++) {
                cur *= nums[j];
                res = Math.Max(res, cur);
            }
        }
        
        return res;
    }
}

File: Data Structures & Algorithms/subsets-ii/submission-10.js
class Solution {
    constructor() {
        this.res = new Set();
    }

    /**
     * @param {number[]} nums
     * @return {number[][]}
     */
    subsetsWithDup(nums) {
        nums.sort((a, b) => a - b);
        this.backtrack(nums, 0, []);
        return Array.from(this.res).map(subset => JSON.parse(subset));
    }

    /**
     * @param {number[]} nums
     * @param {number[]} subset
     * @return {void}
     */
    backtrack(nums, i, subset) {
        if (i === nums.length) {
            this.res.add(JSON.stringify(subset));
            return;
        }

        subset.push(nums[i]);
        this.backtrack(nums, i + 1, subset);
        subset.pop();
        this.backtrack(nums, i + 1, subset);
    }
}